### PR TITLE
Procnet multiple ports

### PR DESCRIPTION
**Pull request by dorkolog**

### CODE REVIEW

File: probe/endpoint/connection_tracker.go [hunk 1]
```diff
@@ -13,11 +13,17 @@ import (
...
 	"github.com/weaveworks/scope/probe/endpoint/procspy"
 	"github.com/weaveworks/scope/report"
 )
 
+var (
+	loopbackIPv4 = net.ParseIP("127.0.0.1")
+	loopbackIPv6 = net.ParseIP("::1")
+)
+
 type connectionTracker struct {
```
Comment (Diffuse2000) [inline probe/endpoint/connection_tracker.go]: MakeIpByte

File: probe/endpoint/connection_tracker.go [hunk 2]
```diff
@@ -165,24 +171,34 @@ func (t *connectionTracker) existingFlows() map[string]fourTuple {
...
+	}
 
-		_, _, incoming := connectionTuple(conn, pidToPort)
+	for _, conn := range connections {
+		incoming := isIncomingConnection(&conn, pidToListenPorts)
 
 		newConn := api.Connection{
 			Proto:   syscall.IPPROTO_TCP,
-			Src:     api.MakeIpByte(conn.LocalAddress), // Unfortunately that the only was it works (for now)
+			Src:     api.MakeIpByte(conn.LocalAddress),
```
Comment (Diffuse2000) [inline probe/endpoint/connection_tracker.go]: MakeIPByte should be called when creating the connections

File: probe/endpoint/connection_tracker.go [hunk 3]
```diff
@@ -197,6 +213,32 @@ func (t *connectionTracker) performWalkProc(rpt *report.Report, hostNodeID strin
...
+	var newConn procspy.Connection
+	newConn.LocalPort = conn.LocalPort
+	newConn.RemotePort = conn.RemotePort
+	newConn.Proc.PID = conn.Proc.PID
+	newConn.Proc.Name = conn.Proc.Name
+	newConn.Proc.NetNamespaceID = conn.Proc.NetNamespaceID
+	newConn.LocalAddress = make(net.IP, len(conn.LocalAddress))
+	newConn.RemoteAddress = make(net.IP, len(conn.RemoteAddress))
+	copy(newConn.LocalAddress, conn.LocalAddress)
+	copy(newConn.RemoteAddress, conn.RemoteAddress)
```
Comment (Diffuse2000) [inline probe/endpoint/connection_tracker.go]: Please put the make and copy lines together, as they are logically part of the same operation. I think it's a good idea to add a function to handle this.

File: probe/endpoint/connection_tracker.go [hunk 4]
```diff
@@ -315,8 +351,21 @@ func connectionTuple(conn *procspy.Connection, listenPorts map[uint]uint16) (fou
 
 	//connection is considered incoming if the port is in listening mode
 	incoming := false
-	if port, ok := listenPorts[conn.Proc.PID]; ok {
-		incoming = port == conn.LocalPort
+	if listenPorts, ok := pidToListenPorts[conn.Proc.PID]; ok {
+		if _, ok := listenPorts[conn.LocalPort]; ok {
+			incoming = true
+		}
 	}
```
Comment (Diffuse2000) [inline probe/endpoint/connection_tracker.go]: You've added a function for doing this operation but it seems like you forgot to use it.

Comment (dorkolog) [reply probe/endpoint/connection_tracker.go]: we don't use ```connectionTuple``` so that's ok

**Pull request merged**